java/client/src/main/java/com/youtube/vitess/client/Proto.java

```diff
@@ -203,6 +210,26 @@ public EntityId apply(Map.Entry<byte[], ?> entry) {
         // Float, Double
         this.type = Query.Type.FLOAT64;
         this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Date) {
+        // Date
+        this.type = Query.Type.DATE;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Time) {
+        // Time
+        this.type = Query.Type.TIME;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Timestamp) {
+        // DateTime, TimeStamp
+        this.type = Query.Type.TIMESTAMP;
+        this.value = ByteString.copyFromUtf8(value.toString());
```

Comment:
I found out that toString() and valueOf() on java.sql.(Date|Time|Timestamp) all use the local machine's default timezone. It'll work out as long as users only ever create java.sql.Timestamp values via valueOf(). But what if they create one from a java.util.Date or `new Timestamp(millis)`? They would have to adjust their epoch times to be intentionally wrong to offset for the local machine's default timezone.

Ideally we would do everything in UTC, but that means we can use neither toString() nor valueOf(). We would have to do the parsing and formatting ourselves. I'm looking into it as part of #1499 where I'm working on the receiving side.